text/2021-10-11-self-made-memory-stress-program.md

```diff
@@ -0,0 +1,27 @@
+# Self-made Memory Stress Program
+
+## Summary
+
+Create a program to replace stress-ng to realize mem stress.
+
+## Motivation
+
+Because stress-ng realize memory stress by applying read and write pressure instead of occupying memory space, which will affect CPU performance, a program purely takes up memory space is need to generate a memory StressChaos.
+
+## Detailed design
+
+Using `syscall.Mmap()` to allocate memory space directly, so that uncertain footprint can be avoided. And write 1 byte on every virtual memory page of the space we allocated can achieve our goal.
+
+```data, err := syscall.Mmap(-1, 0, length, syscall.PROT_READ|syscall.PROT_WRITE, syscall.MAP_PRIVATE|syscall.MAP_ANONYMOUS)```
```

Comment:
```
data, err := syscall.Mmap(-1, 0, length, syscall.PROT_READ|syscall.PROT_WRITE, syscall.MAP_PRIVATE|syscall.MAP_ANONYMOUS)
```

Reply:
Can you add more introduction about `syscall.PROT_READ|syscall.PROT_WRITE, syscall.MAP_PRIVATE|syscall.MAP_ANONYMOUS`

Reply:
The parameter `syscall.PROT_READ|syscall.PROT_WRITE` let us read and write the array `data`, so we can write some bytes in it to occupy memory page.
And the parameter `syscall.MAP_PRIVATE` makes the mapping not visible to other processes and are not carried through to the underlying file, `syscall.MAP_ANONYMOUS` makes the mapping won't be backed by any file. Therefore, the allocated memory will not leave any mark or be accessed.
You can get more information at https://man7.org/linux/man-pages/man2/mmap.2.html.

Reply:
You need to add this introduction on RFC